code/drasil-gool/lib/GOOL/Drasil/LanguageRenderer/SwiftRenderer.hs

```diff
@@ -942,52 +942,83 @@ swiftIndexFunc l v' = do
 swiftMapFunc lst f = objMethodCall (onStateValue valueType lst) lst swiftMap [f]
 
 swiftListAddFunc :: (RenderSym r) => SValue r -> SValue r -> SValue r
 swiftListAddFunc i v = let atArg = var swiftAt int
-  in funcAppMixedArgs swiftListAdd (listType $ onStateValue valueType v) 
+  in funcAppMixedArgs swiftListAdd (listType $ onStateValue valueType v)
     [v] [(atArg, i)]
 
 swiftWriteFunc :: (RenderSym r) => SValue r -> SValue r -> SValue r
 swiftWriteFunc v f = let contentsArg = var swiftContentsOf (obj swiftData)
-  in swiftTryVal $ objMethodCallNamedArgs void f swiftWrite 
-    [(contentsArg, newObj (obj swiftData) [v $. funcFromData (R.func swiftUTF8) 
+  in swiftTryVal $ objMethodCallNamedArgs void f swiftWrite
+    [(contentsArg, newObj (obj swiftData) [v $. funcFromData (R.func swiftUTF8)
     (obj swiftEncoding)])]
 
 swiftReadLineFunc :: (RenderSym r) => SValue r
 swiftReadLineFunc = swiftUnwrapVal $ funcApp swiftReadLine string []
 
 swiftReadFileFunc :: (RenderSym r) => SValue r -> SValue r
-swiftReadFileFunc v = let contentsArg = var swiftContentsOf infile 
+swiftReadFileFunc v = let contentsArg = var swiftContentsOf infile
   in swiftTryVal $ funcAppNamedArgs CP.stringRender' string [(contentsArg, v)]
 
 swiftSplitFunc :: (RenderSym r) => Char -> SValue r -> SValue r
 swiftSplitFunc d s = let sepArg = var swiftSepBy char
   in objMethodCallNamedArgs (listType string) s swiftSplit [(sepArg, litChar d)]
 
 swiftJoinedFunc :: (RenderSym r) => Char -> SValue r -> SValue r
-swiftJoinedFunc d s = let sepArg = var swiftSep char 
+swiftJoinedFunc d s = let sepArg = var swiftSep char
   in objMethodCallNamedArgs string s swiftJoined [(sepArg, litChar d)]
 
 swiftIndexOf :: (RenderSym r) => SValue r -> SValue r -> SValue r
 swiftIndexOf = swiftUnwrapVal .: swiftIndexFunc
 
-swiftListSlice :: (RenderSym r) => SVariable r -> SValue r -> SValue r -> 
-  SValue r -> SValue r -> MSBlock r
-swiftListSlice vn vo beg end step = let i = var "i" int 
-  in block [vn &= swiftMapFunc (swiftStrideFunc beg end step) (lambda [i] 
-    (listAccess vo (valueOf i)))]
-
-swiftPrint :: Bool -> Maybe (SValue SwiftCode) -> SValue SwiftCode -> 
+-- | Swift's syntactic sugar for list slicing.
+--   All but the last two lines are used to manually set the bounds based on
+--   whether `step` is positive or negative, if either bound is not given
+swiftListSlice :: (RenderSym r) => SVariable r -> SValue r ->
+  Maybe (SValue r) -> Maybe (SValue r) -> SValue r -> MSBlock r
+swiftListSlice vn vo beg' end' step = do
+  stepV <- zoom lensMStoVS step
+  let needIf = (isNothing beg' || isNothing end') && isNothing (valueInt stepV)
+  bName <- if isJust beg' || not needIf then pure ""
+           else genVarName [] "begIdx"
+  eName <- if isJust end' || not needIf then pure ""
+           else genVarName [] "endIdx"
+  let -- If beg or end aren't given and step isn't a litInt, calculate them at
+      --  and store in a variable.
```

Comment:
Something seems to be missing from this comment, "calculate them at ... and store in a variable"

Reply:
Good catch!  It's supposed to say "at runtime".  I'll add that back in.